File: src/main/java/kz/superkassa/tests/framework/support/Polling.java
package ru.superkassa.tests.framework.support;

import org.awaitility.Awaitility;
import ru.superkassa.tests.framework.config.TestConfig;

import java.util.concurrent.Callable;

public final class Polling {
    private final TestConfig config;

    public Polling(TestConfig config) {
        this.config = config;
    }

    public <T> T until(Callable<T> supplier) {
        return Awaitility.await()
                .atMost(config.defaultTimeout())
                .pollInterval(config.pollingInterval())
                .until(supplier, value -> value != null);
    }
}

File: src/main/kotlin/kz/superkassa/tests/framework/support/Polling.kt
package kz.superkassa.tests.framework.support

import kz.superkassa.tests.framework.config.TestConfig
import org.awaitility.Awaitility
import java.util.concurrent.Callable

@Suppress("unused")
class Polling(
    private val config: TestConfig,
) {
    fun <T> until(supplier: Callable<T>): T =
        Awaitility.await()
            .atMost(config.defaultTimeout)
            .pollInterval(config.pollingInterval)
            .until(supplier) { it != null }
}
